### ISSUE TITLE
conversion

### ISSUE THREAD

**Issue by tarbear123** [lukefrahm/SettlersOfCatan#2]
how would i convert your code to NET Core?

**Comment by lukefrahm** [comment]
tl;dr you'll have to write code and be familiar with C# development to convert it. If you're not a developer, or not versed in O-O/n-tier architecture, this could be very difficult to implement :)

The .NET framework (the platform for this app) is Windows-only code. .NET core came out not too long ago to more or less allow for C# developers to write universal code that can work OS-agnostic, not that unlike Java. .NET core is much less capable than the Windows framework, though, so not all code can be easily converted

Microsoft has a porting process that can be followed to convert .NET framework to .NET core, but the user interface would need to be rewritten to work on an operating system other than Windows. You could try using Mono to parse and try a port to get the interface (Winforms) to be compatible on platforms other than Windows, but additional development would still be needed with the Linux port for bug fixes from the conversion. https://www.mono-project.com/docs/gui/winforms/porting-winforms-applications/

Assuming you're a developer versed in C style languages, you could likely use both tools (assuming Mono isn't able to help with the logic and data layers) to get the application ported over and then do some fine tuning of the code to ensure stability. Another option under the assumption you're versed in C style languages would be to manually rewrite the application in Java, which natively works with Arduino, but this would require rewriting the entire app. The logic and data layers wouldn't be too terribly difficult, but the interface would be completely incompatible and need to be redeveloped
